vyper/math/math.vy

```diff
@@ -0,0 +1,83 @@
+# In order to precisely calculate ln we are using Arithmetic–geometric mean approach
+# The formula:
+#   ln(x) = pi / 2 * M(1, 4/s) - m * ln(2) where M denotes the arithmetic-geometric mean of 1 and 4/s, and s = x * 2^m
+# ...
+# We have chosen m = 19 => 2^19 = 524288 and 19 * ln(2) = 1316979643063896087892741
+
+
+@public
+@constant
+def big_ln(x: decimal) -> decimal:
+
+    pi: decimal = 31415926535897932384626433832795028.0
+    a: decimal = 100000000000000000000.0
+    # Here we are using (max(1.0 / x, x) because:
+    # if x < 1 => ln(0.1) = -ln(1/0.1) = -ln(10)
+    g: decimal = 400000000000000000000.0 / (max(1.0 / x, x) * 524288.0)
```

Comment:
if you wanted to you could use something like this `convert(40**12, decimal)` this would get constant folded into the same but might be more readable?